File: api/src/main/java/net/blueshell/api/config/JpaEntityListenerBridgeConfig.java
package net.blueshell.api.config;

import net.blueshell.api.common.event.jpa.JpaListener;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.context.annotation.Configuration;

@Configuration
public class JpaEntityListenerBridgeConfig {
    @Autowired
    public JpaEntityListenerBridgeConfig(ApplicationEventPublisher eventPublisher) {
        JpaListener.setPublisher(eventPublisher);
    }
}

File: api/src/main/kotlin/net/blueshell/api/config/JpaEntityListenerBridgeConfig.kt
package net.blueshell.api.config

import net.blueshell.api.listener.base.JpaListener
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationEventPublisher
import org.springframework.context.annotation.Configuration

@Configuration
class JpaEntityListenerBridgeConfig @Autowired constructor(eventPublisher: ApplicationEventPublisher) {
    init {
        JpaListener.setPublisher(eventPublisher)
    }
}